set.mm

```diff
@@ -515900,6 +516500,477 @@ commutative monoid (=vectors) together with a semiring (=scalars) and a
   $}
 
 
+$(
+-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-
+  Subrings generated by a set
+-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-
+$)
+
+  ${
+    elrgspn.b $e |- B = ( Base ` R ) $.
+    elrgspn.m $e |- M = ( mulGrp ` R ) $.
+    elrgspn.x $e |- .x. = ( .g ` R ) $.
+    elrgspn.n $e |- N = ( RingSpan ` R ) $.
+    elrgspn.f $e |- F = { f e. ( ZZ ^m Word A ) | f finSupp 0 } $.
+    elrgspn.r $e |- ( ph -> R e. Ring ) $.
+    elrgspn.a $e |- ( ph -> A C_ B ) $.
+    ${
+      $d .x. a e f g h i j s u v w x y $.  $d A a f g i v w $.  $d A a u $.
+      $d A h i v $.  $d A z $.  $d B a f g v w $.  $d B a u $.  $d B a y $.
+      $d F a f g i v w $.  $d F a u $.  $d F h i v $.  $d F i x y $.  $d F z $.
+      $d M a f g i v w $.  $d M a i y $.  $d M a u $.  $d M h i v $.
+      $d R a f g i v w $.  $d R a u $.  $d R h i v $.  $d R i x y $.
+      $d S g i w x y $.  $d S x z $.  $d g w z $.  $d A a f g i u v w $.
+      $d A h i j t v $.  $d A z $.  $d B a f g u v w $.  $d B a y $.  $d B j $.
+      $d F a f g i u v w $.  $d F h i j t v $.  $d F i x y $.  $d F z $.
+      $d M a f g i u v w $.  $d M a i y $.  $d M h i j v $.
+      $d R a f g i u v w $.  $d R h i j v $.  $d R i x y $.  $d S g i w x y $.
+      $d S x z $.  $d f j t u $.  $d g t w $.  $d g w z $.  $d A f v x $.
+      $d A g v w x $.  $d F g $.  $d M g $.  $d R g v $.  $d R v w $.
+      $d S x $.  $d A e t y $.  $d A f g h i t w x $.  $d A s t $.
+      $d B f g t w $.  $d F e y $.  $d F f g h i w x $.  $d F s $.  $d M e y $.
+      $d M f g h i w x $.  $d M s $.  $d R e t y $.  $d R f g h i t w x $.
+      $d R s t $.  $d S g s t w $.  $d ph a e f g h i j s t u v w x y z $.
```

Comment:
Ususally, we sort the DV-conditions alphabetically (with respect to the class at the beginning) at least.